paimon-core/src/main/java/org/apache/paimon/metrics/MetricGroup.java

```diff
@@ -56,28 +46,19 @@ public interface MetricGroup {
      * @param <T> return type of the gauge
      * @return the given gauge
      */
-    <T, G extends Gauge<T>> G gauge(String name, G gauge);
+    <T> Gauge<T> gauge(String name, Gauge<T> gauge);
```

Comment:
So do we need a wrapper to wrap the PAIMON basic metric types to FLINK basic metrics types? Because finally we need to register FLINK(or SPARK) Metric to FLINK metric groups.

Reply:
Yes. Flink support will come in the next PR.